pennylane/operation.py

```diff
@@ -1151,6 +1151,16 @@ def hyperparameters(self):
         self._hyperparameters = {}
         return self._hyperparameters
 
+    @property
+    def hints(self):
+        """dict: Dictionary of non-trainable variables that don't affect what the operation does,
+        including user tags and compiler hints"""
+        # pylint: disable=attribute-defined-outside-init
+        if hasattr(self, "_hints"):
+            return self._hints
+        self._hints = {}
+        return self._hints
+
```

Comment:
Modifying `Operator` is outside the scope of this PR.

Reply:
@albi3ro agreed! What would be the best alternative here?

Reply:
I removed `hints` and added a different workaround for the issue of operators decomposing to themselves. Only works on length 1 cycle (decompose to self), wouldn't stop any larger chains. At least `equal_if_decomposed` now has an iteration limit so it doesn't get stuck in an infinite loop.